server/reload.go

```diff
@@ -717,33 +734,39 @@ func (s *Server) applyOptions(opts []option) {
 func (s *Server) reloadAuthorization() {
 	s.mu.Lock()
 
+	oldAccounts := s.accounts
+	s.accounts, s.gacc = nil, nil
+	s.configureAccounts()
+
 	s.configureAuthorization()
 
 	// This map will contain the names of accounts that have their streams
 	// import configuration changed.
-	awcsti := make(map[string]struct{}, len(s.opts.Accounts))
+	awcsti := make(map[string]struct{}, len(s.accounts))
 
-	oldAccounts := s.accounts
-	s.accounts = make(map[string]*Account)
-	s.registerAccount(s.gacc)
-	for _, newAcc := range s.opts.Accounts {
+	for _, newAcc := range s.accounts {
 		if acc, ok := oldAccounts[newAcc.Name]; ok {
 			// If account exist in latest config, "transfer" the account's
-			// sublist to the new account object before registering it
-			// in s.accounts.
+			// sublist and client map to the new account.
 			acc.mu.RLock()
 			sl := acc.sl
+			clients := acc.clients
 			acc.mu.RUnlock()
 			newAcc.mu.Lock()
 			newAcc.sl = sl
+			if len(clients) > 0 {
```

Comment:
Technically, a client could be removed from the account after just just capture the pointer, so ranging down below would still be a problem, no?

Reply:
Not understanding. Maybe GH to discuss.